sos/plugins/__init__.py

```diff
@@ -1123,6 +1123,21 @@ def check_process_by_name(self, process):
             return False
         return status
 
+    def get_process_pid(self, process):
+        """Returns a process PID if it exists.
+        If th process doesn't exist, returns -1"""
+        cmd_line_glob = "/proc/[0-9]*/cmdline"
+        cmd_line_paths = glob.glob(cmd_line_glob)
+        for path in cmd_line_paths:
+            try:
+                with open(path, 'r') as f:
+                    cmd_line = f.read().strip()
+                    if process in cmd_line:
+                        return path.split("/")[2]
+            except IOError as e:
+                continue
+        return -1
```

Comment:
I think it would be better to use `None` as the return type here. In some ways, in Python style, an exception would be preferred (rather than signalling the error with a return value), but I think that is overkill here - it would force us to wrap all those calls in `try`/`except` blocks.

The problem with `-1` is twofold: the test looks less obvious (`if foo != -1` vs. `if not foo` of `if foo is not None`). The second form of the `None` comparison is only needed if we want to distinguish from `0` - I'm inclined to think that's not necessary since `pid 0` is the swapper thread, and we don't really want to do anything with that special value. The other issue with `-1` is that it is a special value recognised by many process tools as well as syscalls: in the signal sending syscalls, as well as the `kill` program, `-1` means "all processes with a PID larger than 1 are signaled". I know that isn't going to cause trouble for your use but it could become relevant in other situations.